### PR TITLE
[3/N] Sets up testing for repos

### PR DESCRIPTION
**Pull request by saurav-c**
## Describe your changes and why you are making these changes
This PR sets up database integration tests for all repos.

The key decisions made are:
- Moving all tests to `repos/tests`
- Each repo should have a corresponding test file named, e.g. `workflow.go`. Inside this test file will be a test case for each repo method, such as `TestWorkflow_Exists`, `TestWorkflow_Get`.
- Any `seed` functions, such as `seedWorkflow` or `seedUser` or `seedDAGWithWorkflow` should be implemented in `seed.go`. These are essentially helper methods that each test can call to quickly generate some test data.
- The database will be wiped after EACH test case. This follows the current approach we have. This is because we just want to verify that each query is correct.

## Checklist before requesting a review
- [ ] I have created a descriptive PR title. The PR title should complete the sentence "This PR...".
- [ ] I have performed a self-review of my code.
- [ ] I have included a small demo of the changes. For the UI, this would be a screenshot or a Loom video.
- [ ] If this is a new feature, I have added unit tests and integration tests.
- [ ] I have run the integration tests locally and they are passing.
- [ ] I have run the linter script locally (See `python3 scripts/run_linters.py -h` for usage).
- [ ] All features on the UI continue to work correctly.
- [ ] Added one of the following CI labels:
    - `run_integration_test`: Runs integration tests
    - `skip_integration_test`: Skips integration tests (Should be used when changes are ONLY documentation/UI)

### CODE REVIEW

File: src/golang/lib/repos/tests/suite.go [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+		return err
+	}
+
+	return nil
+}
+
+// TearDownTest is run after each test finishes.
+func (ts *TestSuite) TearDownTest() {
+	// Clear all of the tables
+	query := `
```
Comment (eunice-chan) [inline src/golang/lib/repos/tests/suite.go:78]: Does this delete everything in the tables? I wonder if there is some way to just delete anything created from the tests..

**Review by eunice-chan — APPROVED** [review]

File: src/golang/lib/repos/tests/suite.go (hunk 1 above)
Comment (saurav-c) [reply src/golang/lib/repos/tests/suite.go:78]: Yeah, this deletes everything, which technically is only things created by the tests since the db is cleared after each test.

File: src/golang/lib/models/schema.go [hunk 2]
```diff
@@ -0,0 +1,4 @@
+package models
+
+// SchemaVersion is the current database schema version.
```
Comment (kenxu95) [inline src/golang/lib/models/schema.go:3]: Drive-by-comment from your slack message :P Can you put a big capital note here that this is the source of truth? (with whatever caveats there are)

Comment (saurav-c) [reply src/golang/lib/models/schema.go:3]: Sure will add in a diff related PR I'm about to merge.
